### ISSUE TITLE
Improve feedback on component pushing

### ISSUE THREAD

**Issue by jorgemoralespou** [redhat-developer/odo#829]
With the supervisord merged, when you are in the development cycle you might end up in situations where you're not sure what's going on behind the scenes. It would be good to provide appropriate feedback to the user, so he can take decisions based on that feedback.

e.g. Create and immediately push.

```bash
$ odo create wildfly --binary=target/ROOT.war
Component 'wildfly' was created and port 8080/TCP was opened
To push source code to the component run 'odo push'

Component 'wildfly' is now set as active component.
$ odo push
Pushing changes to component: wildfly-binary  # THIS MESSAGE TAKES A WHILE
Please wait, building component....
...
changes successfully pushed to component: wildfly-binary
```

Push will be blocked for long time before output any message, waiting for the deployment to be ready.

Instead it would be better to have something like:
```bash
$ odo create wildfly --binary=target/ROOT.war
Component 'wildfly' was created and port 8080/TCP was opened
To push source code to the component run 'odo push'

Component 'wildfly' is now set as active component.
$ odo push
Component 'wildfly' is not yet ready on the platform. It's being deployed.
Component 'wildfly' is now reading.
Pushing changes to component: wildfly 
Please wait, building component....
...
changes successfully pushed to component: wildfly
```

**NOTE:** The message added are the 2 lines after `odo push`.

**Comment by cdrage** [comment]
I've taken this up and pushed it to https://github.com/redhat-developer/odo/pull/917

**Comment by kadel** [comment]
closed in #917
